src/_pytest/config/__init__.py

```diff
@@ -1268,25 +1267,42 @@ def _getini(self, name: str):
         #     a_line_list = ["tests", "acceptance"]
         #   in this case, we already have a list ready to use
         #
-        if type == "pathlist":
+        if ini_type == "pathlist":
             # TODO: This assert is probably not valid in all cases.
             assert self.inifile is not None
             dp = py.path.local(self.inifile).dirpath()
+            if not value:  # we need this because otherwise mypy raises an error
+                value = []
             input_values = shlex.split(value) if isinstance(value, str) else value
             return [dp.join(x, abs=True) for x in input_values]
-        elif type == "args":
+        elif ini_type == "args":
             return shlex.split(value) if isinstance(value, str) else value
-        elif type == "linelist":
+        elif ini_type == "linelist":
             if isinstance(value, str):
                 return [t for t in map(lambda x: x.strip(), value.split("\n")) if t]
             else:
                 return value
-        elif type == "bool":
+        elif ini_type == "bool":
             return _strtobool(str(value).strip())
         else:
-            assert type is None
+            assert ini_type is None
             return value
 
+    def _getini(self, name: str):
```

Comment:
BTW, I noticed that `getini` is just `_getini` with `@lru_cache(None)` (which will also be faster) -- if you feel like doing this as a follow-up.